### DISCUSSION TITLE
Using entity relation extraction to establish an entity hierarchy

### DISCUSSION THREAD

**Discussion by F1nalLap** [explosion/spaCy#11128] [Help: Model Advice]
Hello, I’m looking to extract entity relations in spacy. For my use-case, I want to label two types of relations from text involving chess matches. I wish to relate a PERSON entity to a CHESS_PIECE entity with the relation label ‘CONTROLS_PIECE’’ and label the hierarchy of CHESS_PIECE values as ‘VALUE_ABOVE’.

For example:

Bobby Fischer moved white knight to a3. Garry Kasparov moved dark pawn to b6.

Entity Links:

(Bobby Fischer, white knight, ‘CONTROLS_PIECE’)

(Garry Kasparov, dark pawn, ‘CONTROLS_PIECE’)

(white knight, dark pawn, ‘VALUE_ABOVE’)

So far I have a custom NER that can recognize people and chess pieces. I’ve been using the rel_component tutorial (https://www.youtube.com/watch?v=8HL-Ap5_Axo) as a starting point and have annotated some relevant examples, however the model does not seem to be learning how to assign the ‘VALUE_ABOVE’ relation. Outside of annotating more data, is there a way I can improve the component’s ability to recognize this relation? If this is above the capabilities of the relation component, should I try to use a matcher instead?

**Comment by polm** [comment]
What is the `VALUE_ABOVE` relation? Is it based on point values for pieces?

If so I don't think it makes sense to train a model for that. The relation is defined simply by the kind of piece, so you can just write rules for it, and besides the piece names there's no information in your input text that you could derive the information from.

**Reply by F1nalLap** [reply]
Thank you for the response, and yes VALUE_ABOVE is based on the point values for each piece. Since the knight is worth 3 points and a pawn is 1 point, I wish to apply the tag VALUE_ABOVE between the knight and the pawn. By rules you are referring to writing a Token/Phrase/Dependency Matcher rule as described here (https://spacy.io/usage/rule-based-matching#matcher) correct?

**Reply by polm** [reply]
No, in this case I'm not referring to the Matchers. Instead you could have a custom component where you just use code to identify pieces and check their relation. I guess you could use an EntityRuler with IDs to assign piece types.
